### PR TITLE
Feature/theme system

### PR DESCRIPTION
**Pull request by kyledoesdev**
Theme System, Manager, Provider, Facade and tests.

Theme Manager has
-> register
->get
->getAll

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     ],
     "require": {
         "illuminate/collections": "^10.0",
-        "illuminate/support": "^10.1"
+        "illuminate/support": "^10.3"
```
Comment (aidanscode) [inline composer.json:18]: Why the version bump?

File: src/Themes/Theme.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Carnival\Themes;
+
+interface Theme {
+    
+    function getName() : string;
+
+    function getDescription() : string;
+
+    function getLogo() : string;
```
Comment (aidanscode) [inline src/Themes/Theme.php]: I think we should leave out logos for now until we figure out how we want to handle images.
At the very least, it isn't obvious exactly what `getLogo` is returning in this state

File: src/Themes/Theme.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Carnival\Themes;
+
+interface Theme {
```
Comment (aidanscode) [inline src/Themes/Theme.php:5]: I forget if we discussed it on the call, but what do you think about a `getVersion` method?
Most themes will come in a plugin so their version number will likely be the version of the plugin, but at least the official theme won't come in a plugin so we can't rely on that relationship.

File: src/Themes/ThemeManager.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    public function __construct() {
+        $this->themes = collect();
+    }
+
+    function register(Theme $theme) : void {
+        $this->themes->put(get_class($theme), $theme);
+    }
+
+    function get(string $themeClass) : Theme {
```
Comment (aidanscode) [inline src/Themes/ThemeManager.php]: What happens here if there is no theme under the requested `$themeClass`?
Perhaps this should return an optional?

File: tst/Carnival/Providers/ThemeServiceProviderTest.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    
+    public function testThemeServiceProviderCanRegisterThemeManagerFacade() {
+        $application = Mockery::mock(Application::class);
+        $themeServiceProvider = new ThemeServiceProvider($application);
+        $application->shouldReceive('bind')->with(
+            'themes', 
+            Mockery::on(fn ($argument) => is_callable($argument) && $argument() instanceof ThemeManager )
+        )->once();
+        $themeServiceProvider->register();
+    }
```
Comment (aidanscode) [inline tst/Carnival/Providers/ThemeServiceProviderTest.php:20]: [Space Police]

There's whitespace between the this method and the top of the class but not the bottom

File: tst/Carnival/Themes/ThemeManagerTest.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    public function testCanGetThemeFromThemeManagerThatIsRegistered() {
+        $this->assertFalse($this->themeManager->getAll()->contains($this->exampleTheme));
+        $this->registerExampleTheme();
+        $this->assertEquals($this->exampleTheme, $this->themeManager->get(get_class($this->exampleTheme)));
+    }
+
+    private function registerExampleTheme() {
+        $this->themeManager->register($this->exampleTheme);
+    }
+}
```
Comment (aidanscode) [inline tst/Carnival/Themes/ThemeManagerTest.php:41]: [Space Police]
Here too

**Review by aidanscode — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (kyledoesdev) [reply composer.json:18]: i dont even remember changing this it mustve been when I ran a command at some point

File: src/Themes/Theme.php (hunk 2 above)
Comment (kyledoesdev) [reply src/Themes/Theme.php]: Okay, I'll remove it for now

File: src/Themes/Theme.php (hunk 3 above)
Comment (kyledoesdev) [reply src/Themes/Theme.php:5]: Sure, I'll add a getVersion() method

File: src/Themes/ThemeManager.php (hunk 4 above)
Comment (kyledoesdev) [reply src/Themes/ThemeManager.php]: Okay I'll change it to return ?Theme

**Review by aidanscode — APPROVED** [review]
Looks nice, thanks!

**Pull request merged**